### DISCUSSION TITLE
zoom timeline, centred on scrub handlebar

### DISCUSSION THREAD

**Discussion by ElectronicBattle** [blakeblackshear/frigate#13088] [Ideas]
Hello 

I searched with no success  using the github "[reminder](https://github.com/blakeblackshear/frigate/discussions?discussions_q=zoom+timeline%2C+centred+on+scrub+handlebar)" and separately via  search terms of `zoom` `timeline` `wheel` `mousewheel` which also failed to find what I wanted to suggest. My idea is perhaps nominally similar to some aspects of [this earlier idea](https://github.com/blakeblackshear/frigate/discussions/12255). 

From the current 0.14 UI, this is the standard timeline with the handlebar more-or-less in the centre:
![image](https://github.com/user-attachments/assets/83c165ed-888c-49af-9ff9-d1731f03d3bc)

UI improvement idea:

1. Using a mouse: hover (preferably, or alternatively click if hovering isn't feasible) on the handlebar. Use the mouse wheel to zoom in/out on the timeline so that instead of the top and bottom representing nominally +/- 25 minutes as per the current UI design, the scale could now arbitrarily represent +/- <<25 minutes (but bounded by sensible limits).
 
2. Using a mouse or touchscreen gesture: shift-click or pinch-and-zoom on the timeline to achieve the same effect (in each case, it could either be achieved symmetrically about the handlebar or independent).

This is not frame-by-frame analysis, something which is not feasible in browser stream inter-process communications as far as I have understood, but it might be an attractive way to implement closer scrutiny.

The 0.14 UI has clearly had a great deal of thought and expertise applied to it - successfully so - and perhaps this sort of notion has already been considered and dismissed. Please feel free to close this idea without response if so!

**Comment by hawkeye217** [comment]
There is already a feature request for this here: https://github.com/blakeblackshear/frigate/issues/11398

This will be coming in a later version.

**Reply by ElectronicBattle** [reply]
good to see 11398 but frustrating that it eluded me despite my ten minutes of diligent searching. How did you search for it, or did you remember it from before?

In any case - if you can, please delete this idea (I can't see how to do so) or if that's not possible, I'll edit my post to point to @NickM-27 (and perhaps I'll edit that thread to point back at this one).

**Reply by hawkeye217** [reply]
No problem. I wrote the timeline code and I remembered the feature request. I'll just close this.

**Reply by ElectronicBattle** [reply]
thanks!
